packages/components/src/components/report-list/ReportList.tsx

```diff
@@ -97,78 +92,159 @@ class ReportMetadata extends React.PureComponent<ReportConsumer> {
-                        </div>
+            <div className="report-list__unsupported-preview">
+                <div className="alert alert-warning unsupported-alert">
+                    <div className="unsupported-alert__icon">
+                        <span className="fa fa-exclamation-circle"/>
                     </div>
 
-                    <ReportMetadata report={this.props.report} />
+                    <p className="unsupported-alert__message">
+                        This report is not currently supported. It is recommended that you view the report in
+                        LabKey Server.
+                    </p>
+
+                    <div className="unsupported-alert__view-link">
+                        <a href={this.props.report.runUrl} className="btn btn-warning">View in LabKey</a>
+                    </div>
                 </div>
-            </Modal.Body>
+
+                <ReportMetadata report={this.props.report} />
+            </div>
         );
     }
 }
 
-class GridReportBody extends React.PureComponent<ReportConsumer> {
+class GridReportBody extends PureComponent<ReportConsumer> {
     render () {
         const { schemaName, queryName, viewName, runUrl, appUrl } = this.props.report;
         const schemaQuery = SchemaQuery.create(schemaName, queryName, viewName);
 
         return (
-            <Modal.Body>
-                <div className="report-list__grid-preview">
-                    <ReportLinks report={this.props.report} />
+            <div className="report-list__grid-preview">
+                <ReportLinks report={this.props.report} />
 
-                    <PreviewGrid schemaQuery={schemaQuery} numCols={4} numRows={3} />
+                <PreviewGrid schemaQuery={schemaQuery} numCols={4} numRows={3} />
 
-                    <ReportMetadata report={this.props.report} />
-                </div>
-            </Modal.Body>
+                <ReportMetadata report={this.props.report} />
+            </div>
         );
     }
 }
 
-class ChartReportBody extends React.PureComponent<ReportConsumer, any> {
+class ChartReportBody extends PureComponent<ReportConsumer, any> {
     render() {
         return (
-            <Modal.Body>
-                <div className="report-list__chart-preview">
-                    <ReportLinks report={this.props.report} />
+            <div className="report-list__chart-preview">
+                <ReportLinks report={this.props.report} />
+
+                <Chart chart={new DataViewInfo(this.props.report)} />
 
-                    <Chart chart={new DataViewInfo(this.props.report)} />
+                <ReportMetadata report={this.props.report} />
+            </div>
+        );
+    }
+}
 
-                    <ReportMetadata report={this.props.report} />
+interface SampleComparisonState {
+    loading?: boolean,
+    report?: any,
+    error?: string,
+}
+
+class SampleComparisonReportBody extends PureComponent<ReportConsumer, SampleComparisonState> {
```

Comment:
Since this is specific to a single chart type (i.e. SCR), should this be split out of this ReportList.tsx file and into its own, something like SampleComparisonReport.tsx?

Reply:
I don't think it's worth the added complexity. It's not useable outside of this file, and won't be, plus we already have three other modal types in this file (Unsupported, Grid, Chart), and this entire file is still under 500 lines long. If I split this into its own file I would also need to add a third file for the interfaces that these components share in order to not have circular dependencies.